### ISSUE TITLE
Intermittent failure in /_mozilla/mozilla/iframe_contentDocument.html

### ISSUE THREAD

**Issue by jdm** [servo/servo#11036]
```
 ▶ Unexpected subtest result in /_mozilla/mozilla/iframe_contentDocument.html:
  │ FAIL [expected PASS] Iframe contentDocument test.
  │   → iframe.contentDocument.getElementById(...) is null
  │ 
  │ test_contentWindow@http://web-platform.test:8000/_mozilla/mozilla/iframe_contentDocument.html:11:19
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1397:20
  └ Test.prototype.step_func/&lt;@http://web-platform.test:8000/resources/testharness.js:1421:20
```
This test was written before the load event for iframes was implemented. It should be rewritten to use that instead of using setTimeout.

Code: `tests/wpt/mozilla/tests/mozilla/iframe_contentDocument.html`
Test: `./mach test-wpt tests/wpt/mozilla/tests/mozilla/iframe_contentDocument.html`

**Comment by yodalee** [comment]
I think I can take this. Here is my current understand:
Instead of check `!iframe.contentWindow` and set a timeout function to check it again, we just need to set `onload` function of iframe, and put `assertEquals` inside. Something like:
```
iframe.onload = function() {
  assert_equals(iframe.contentDocument.getElementById('test').textContent, 'value');
  this.done();
};
```

**Comment by jdm** [comment]
You'll need
``js
iframe.onload = this.step_func_done(function() {
    assert_equals(...);
})
```
to avoid the timeout.

**Comment by ConnorGBrewster** [comment]
Fixed in #11053
